pkg/consuldp/xds.go

```diff
@@ -73,7 +73,9 @@ func (cdp *ConsulDataplane) setupXDSServer() error {
 	// However, we needed this fix (https://github.com/mwitkow/grpc-proxy/pull/62) which was available on the fork we are using.
 	// TODO: Switch to the main library once the fix is merged to keep upto date.
 	newGRPCServer := grpc.NewServer(
-		grpc.UnknownServiceHandler(proxy.TransparentHandler(cdp.director)),
+		// Increase the maximum message size due to large proxies sometimes exceeding the default 4MB limit.
+		grpc.MaxRecvMsgSize(50*1024*1024),
```

Comment:
Maybe move this 50MB into a constant since it's a magic number and you use it twice?